src/surrogate_heat_driver.cpp

```diff
@@ -87,24 +164,148 @@ void SurrogateHeatDriver::generate_arrays()
   // Make a radial grid for the fuel with equal spacing.
   r_grid_fuel_ = xt::linspace<double>(0, pellet_radius_, n_fuel_rings_ + 1);
 
-  // Create empty arrays for source term and temperature
+  // Compute the cross-sectional areas of each region
+  solid_areas_ = xt::empty<double>({n_rings()});
+  for (int i = 0; i < n_rings(); ++i) {
+    if (i < n_fuel_rings_)
+      solid_areas_(i) = M_PI * (r_grid_fuel_(i + 1) * r_grid_fuel_(i + 1) - r_grid_fuel_(i) * r_grid_fuel_(i));
+    else {
+      int r = i - n_fuel_rings_;
+      solid_areas_(i) = M_PI * (r_grid_clad_(r + 1) * r_grid_clad_(r + 1) - r_grid_clad_(r) * r_grid_clad_(r));
+    }
+  }
+
+  // Create empty arrays for source term and temperature in the solid phase
   source_ = xt::empty<double>({n_pins_, n_axial_, n_rings()});
   temperature_ = xt::empty<double>({n_pins_, n_axial_, n_rings()});
   density_ = xt::zeros<double>({n_pins_, n_axial_, n_rings()});
   fluid_mask_ = xt::zeros<int>({n_pins_, n_axial_, n_rings()});
+
+  // Create empty arrays for temperature and density in the fluid phase
+  fluid_temperature_ = xt::empty<double>({n_pins_, n_axial_});
+  fluid_density_ = xt::empty<double>({n_pins_, n_axial_});
 }
 
 void SurrogateHeatDriver::solve_step()
 {
-  std::cout << "Solving heat equation...\n";
-  // NuScale inlet temperature
-  double T_co = 523.15;
+  solve_fluid();
+  solve_heat();
+}
 
-  // Set initial temperature
-  for (auto& T : temperature_) {
-    T = T_co;
+void SurrogateHeatDriver::solve_fluid()
+{
+  // convenience function for determining the rod power in a given axial layer
+  auto rod_axial_node_power = [this](int pin, int axial) {
+    Expects(axial <= n_axial_);
+
+    double power = 0.0;
+    double dz = z_(axial + 1) - z_(axial);
+
+    for (int i = 0; i < n_rings(); ++i)
+      power += source_(pin, axial, i) * solid_areas_(i) * dz;
+
+    return power;
+  };
+
+  // determine the power deposition in each channel; the target applications will always
+  // be steady-state or pseudo-steady-state cases with no axial conduction such that
+  // the power deposition in each channel is independent of a convective heat transfer
+  // coefficient and only depends on the rod power at that axial elevation. The channel
+  // powers are indexed by channel ID, axial ID
+  xt::xtensor<double, 2> channel_powers = xt::zeros<double>({n_channels_, n_axial_});
```

Comment:
I think you can use the following constructor directly
```suggestion
  xt::xtensor<double, 2> channel_powers({n_channels_, n_axial_}, 0.0);
```

Reply:
Looks like you missed this one?